Data/core_rulebook/cr_skills.lua

```diff
@@ -0,0 +1,8771 @@
+    },
+    {
+      Category="SKILL",
+      Variables={
+        "Craft (Stonemasonry)",
+      },
+      Formula=Formula("6"),
+      Type={
+        Name="SkillFocus",
+        Replace=false,
+        Stack=false,
+      },
+      Conditions={
+        function (character)
+          return 1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Skill Focus (Craft (Stonemasonry))")
+          end)
+        end,
+        function (character)
+          return ((character.Skill("Craft (Stonemasonry)").ranks >= 10 and 1 or 0)) >= 1
+        end,
+      },
+    },
+  },
+})
+ModifySkill({
+  Name="Craft (Traps)",
+  Bonuses={
+    {
+      Category="SKILL",
+      Variables={
+        "Craft (Traps)",
+      },
+      Formula=Formula("3"),
+      Type={
+        Name="SkillFocus",
+        Replace=false,
+        Stack=false,
+      },
+      Conditions={
+        function (character)
+          return 1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Skill Focus (Craft (Traps))")
+          end)
+        end,
+      },
+    },
+    {
+      Category="SKILL",
+      Variables={
+        "Craft (Traps)",
+      },
+      Formula=Formula("6"),
+      Type={
+        Name="SkillFocus",
+        Replace=false,
+        Stack=false,
+      },
+      Conditions={
+        function (character)
+          return 1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Skill Focus (Craft (Traps))")
+          end)
+        end,
+        function (character)
+          return ((character.Skill("Craft (Traps)").ranks >= 10 and 1 or 0)) >= 1
+        end,
+      },
+    },
+  },
+})
+ModifySkill({
+  Name="Craft (Untrained)",
+  Bonuses={
+    {
+      Category="SKILL",
+      Variables={
+        "Craft (Untrained)",
+      },
+      Formula=Formula("3"),
+      Type={
+        Name="SkillFocus",
+        Replace=false,
+        Stack=false,
+      },
+      Conditions={
+        function (character)
+          return 1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Skill Focus (Craft (Untrained))")
+          end)
+        end,
+      },
+    },
+    {
+      Category="SKILL",
+      Variables={
+        "Craft (Untrained)",
+      },
+      Formula=Formula("6"),
+      Type={
+        Name="SkillFocus",
+        Replace=false,
+        Stack=false,
+      },
+      Conditions={
+        function (character)
+          return 1 <= #filter(character.Abilities, function (ability)
+            return ability.Category == "FEAT" and (ability.Name == "Skill Focus (Craft (Untrained))")
+          end)
+        end,
+        function (character)
+          return ((character.Skill("Craft (Untrained)").ranks >= 10 and 1 or 0)) >= 1
+        end,
+      },
+    },
+  },
+})
+ModifySkill({
```

Comment:
Really?  The SkillFocus is built into every single skill instead of just being something on the feat?  These files make me so sad...  This seems like this would be an easy ability that could have been a simple CHOOSE:SKILL on the feat with a matched BONUS

Reply:
yep... its bad.